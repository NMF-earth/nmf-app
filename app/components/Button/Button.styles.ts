import { StyleSheet } from "react-native";

import { Colors } from "style";

const Primary = StyleSheet.create({
  default: {
    backgroundColor: Colors.primary,
  },
});

const Secondary = StyleSheet.create({
  default: {
    backgroundColor: Colors.secondary,
  },
});

const Danger = StyleSheet.create({
  default: {
    backgroundColor: Colors.danger,
  },
});

export default { Primary, Secondary, Danger };
