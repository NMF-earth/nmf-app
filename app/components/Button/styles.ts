import { StyleSheet } from "react-native";

import { Colors } from "style";

export default StyleSheet.create({
  default: {
    paddingHorizontal: 36,
  },
  black: {
    backgroundColor: Colors.grey100,
  },
  fullWidth: {
    width: "100%",
    alignItems: "center",
    paddingHorizontal: 0,
  },
});
