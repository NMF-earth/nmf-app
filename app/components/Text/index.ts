import H1 from "./H1";
import H2 from "./H2";
import H3 from "./H3";
import Primary from "./Primary";
import Secondary from "./Secondary";
import Tertiary from "./Tertiary";
import Link from "./Link";

export default { H1, H2, H3, Primary, Secondary, Tertiary, Link };
