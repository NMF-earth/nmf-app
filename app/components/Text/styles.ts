import { StyleSheet } from "react-native";
import colors from "../../style/colors";
import { fontWeight } from "../../style/font";

export default StyleSheet.create({
  darkGray: {
    color: colors.darkLink70
  },
  lightGray: {
    color: colors.swordGray
  },
  green: {
    color: colors.linkGreen
  },
  white: {
    color: colors.white
  },
  bold: {
    fontFamily: fontWeight.Bold
  },
  center: {
    textAlign: "center"
  }
});
