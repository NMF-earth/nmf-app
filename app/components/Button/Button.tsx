import React from "react";
import { TouchableOpacity, StyleProp, ViewStyle } from "react-native";
import { Ionicons } from "@expo/vector-icons";

import { Colors } from "style";

import Text from "../Text";
import styles from "./Button.styles";
import mainStyle from "./styles";

const PADDING_VERTICAL = 12;

type ButtonType = "Primary" | "Secondary" | "Danger";

type Props = {
  style?: StyleProp<ViewStyle>;
  fullWidth?: boolean;
  onPress: () => void;
  text: string;
  icon?: string;
};

interface ButtonFactory {
  (type: ButtonType): React.FC<Props>;
}

const buttonFactory: ButtonFactory = (type) => (props) => {
  const style = [mainStyle.default, styles[type].default, props.style];
  const { fullWidth, onPress, text, icon } = props;
  let iconItem = null;

  if (fullWidth) {
    style.push(mainStyle.fullWidth);
  }

  if (icon) {
    iconItem = (
      <Ionicons
        name={icon as keyof typeof Ionicons.glyphMap}
        size={24}
        style={mainStyle.mainIcon}
        color={Colors.white}
      />
    );
  }

  const additionalStyle = {
    paddingVertical: PADDING_VERTICAL,
    borderRadius: PADDING_VERTICAL * 2,
  };
  return (
    <TouchableOpacity onPress={onPress} {...props} style={[style, additionalStyle]}>
      <Text.Primary bold center white>
        {text}
      </Text.Primary>
      {iconItem}
    </TouchableOpacity>
  );
};

const Primary = buttonFactory("Primary");
const Secondary = buttonFactory("Secondary");
const Danger = buttonFactory("Danger");

export default { Primary, Secondary, Danger };
