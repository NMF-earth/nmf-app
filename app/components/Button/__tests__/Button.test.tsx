import React from "react";
import { create } from "react-test-renderer";

import Button from "../";
import Text from "../../Text";

jest.unmock("../");

const foo = () => {
  // do nothing.
};

/* Primary Button Tests */

it("renders correctly full width Primary Button with Primary font", () => {
  const tree = create(
    <Button.Primary fullWidth onPress={foo} textType={"Primary"}>
      <Text.Primary>test</Text.Primary>
    </Button.Primary>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Primary Button with Secondary font", () => {
  const tree = create(
    <Button.Primary onPress={foo} textType={"Secondary"}>
      <Text.Secondary>test</Text.Secondary>
    </Button.Primary>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

/* Secondary Button Tests */

it("renders correctly full width Secondary Button with Primary font", () => {
  const tree = create(
    <Button.Secondary fullWidth onPress={foo} textType={"Primary"}>
      <Text.Primary>test</Text.Primary>
    </Button.Secondary>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Secondary Button with Secondary font", () => {
  const tree = create(
    <Button.Secondary onPress={foo} textType={"Secondary"}>
      <Text.Secondary>test</Text.Secondary>
    </Button.Secondary>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
