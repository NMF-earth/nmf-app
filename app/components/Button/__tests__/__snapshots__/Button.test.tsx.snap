// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Primary Button with Secondary font 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    Array [
      Array [
        Object {
          "paddingHorizontal": 42,
        },
        Object {
          "backgroundColor": "#4BB377",
        },
        undefined,
      ],
      Object {
        "borderRadius": 40,
        "paddingVertical": 12,
      },
    ]
  }
  textType="Secondary"
>
  <Text.Secondary>
    test
  </Text.Secondary>
</TouchableOpacity>
`;

exports[`renders correctly Primary Button with Tertiary font 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    Array [
      Array [
        Object {
          "paddingHorizontal": 42,
        },
        Object {
          "backgroundColor": "#4BB377",
        },
        undefined,
      ],
      Object {
        "borderRadius": 38,
        "paddingVertical": 12,
      },
    ]
  }
  textType="Tertiary"
>
  <Text.Tertiary>
    test
  </Text.Tertiary>
</TouchableOpacity>
`;

exports[`renders correctly Secondary Button with Secondary font 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    Array [
      Array [
        Object {
          "paddingHorizontal": 42,
        },
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#4BB377",
          "borderWidth": 2.5,
        },
        undefined,
      ],
      Object {
        "borderRadius": 40,
        "paddingVertical": 12,
      },
    ]
  }
  textType="Secondary"
>
  <Text.Secondary>
    test
  </Text.Secondary>
</TouchableOpacity>
`;

exports[`renders correctly Secondary Button with Tertiary font 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    Array [
      Array [
        Object {
          "paddingHorizontal": 42,
        },
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#4BB377",
          "borderWidth": 2.5,
        },
        undefined,
      ],
      Object {
        "borderRadius": 38,
        "paddingVertical": 12,
      },
    ]
  }
  textType="Tertiary"
>
  <Text.Tertiary>
    test
  </Text.Tertiary>
</TouchableOpacity>
`;

exports[`renders correctly full width Primary Button with Primary font 1`] = `
<TouchableOpacity
  fullWidth={true}
  onPress={[Function]}
  style={
    Array [
      Array [
        Object {
          "paddingHorizontal": 42,
        },
        Object {
          "backgroundColor": "#4BB377",
        },
        undefined,
        Object {
          "alignItems": "center",
          "paddingHorizontal": 0,
          "width": "100%",
        },
      ],
      Object {
        "borderRadius": 42,
        "paddingVertical": 12,
      },
    ]
  }
  textType="Primary"
>
  <Text.Primary>
    test
  </Text.Primary>
</TouchableOpacity>
`;

exports[`renders correctly full width Secondary Button with Primary font 1`] = `
<TouchableOpacity
  fullWidth={true}
  onPress={[Function]}
  style={
    Array [
      Array [
        Object {
          "paddingHorizontal": 42,
        },
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#4BB377",
          "borderWidth": 2.5,
        },
        undefined,
        Object {
          "alignItems": "center",
          "paddingHorizontal": 0,
          "width": "100%",
        },
      ],
      Object {
        "borderRadius": 42,
        "paddingVertical": 12,
      },
    ]
  }
  textType="Primary"
>
  <Text.Primary>
    test
  </Text.Primary>
</TouchableOpacity>
`;
